File: C++/229. Majority Element II.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int n = nums.size();
        vector<int> res;
        map<int,int> map;
        for(int i=0; i<n; i++){
            map[nums[i]]++;
            if(map[nums[i]]>n/3){
                res.push_back(nums[i]);
                map[nums[i]]=INT_MIN;
            }
        }
        return res;
    }
};

File: Java/229. Majority Element II.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length;
        List<Integer> res = new ArrayList<>();
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0; i<n; i++){
            map.put(nums[i], map.getOrDefault(nums[i],0)+1);
            if(map.get(nums[i])>n/3){
                res.add(nums[i]);
                map.put(nums[i],Integer.MIN_VALUE);
            }
        }
        return res;
    }
}
